File: src/main/java/com/example/stopgap/exception/ExceptionConfig.java
package com.example.stopgap.exception;

import com.example.stopgap.instanceregistry.InstanceRegistry;

public final class ExceptionConfig {
    private ExceptionConfig() {
    }

    public static void setup(final InstanceRegistry registry) {
        registry.registerForType(ExceptionEndpoint.class, ExceptionConfig::exceptionEndpoint);
    }

    private static ExceptionEndpoint exceptionEndpoint(final InstanceRegistry registry) {
        return new ExceptionEndpoint();
    }
}

File: src/main/kotlin/com/example/stopgap/exception/ExceptionConfig.kt
package com.example.stopgap.exception

import com.example.stopgap.instanceregistry.InstanceRegistry

object ExceptionConfig {

    fun setup(registry: InstanceRegistry) {
        registry.registerForType(::exceptionEndpoint)
    }

    private fun exceptionEndpoint(registry: InstanceRegistry): ExceptionEndpoint {
        return ExceptionEndpoint()
    }

}
